lib/caching-precompiler.js

```diff
@@ -47,8 +48,33 @@ CachingPrecompiler.prototype._init = function () {
 CachingPrecompiler.prototype._transform = function (code, filePath, hash) {
 	code = code.toString();
 
-	var options = babelConfigHelper.build(this.babelConfig, this.powerAssert, filePath, code);
-	var result = this.babel.transform(code, options);
+	var fileExt = path.extname(filePath).replace(/^\./, '');
+	var isCustomExt = this.extensions.filter(function (ext) {
+		return (ext !== 'js' && ext === fileExt);
+	}).length > 0;
+
+	var result;
+
+	if (isCustomExt) {
+		switch (fileExt) {
```

Comment:
@vdemedes Agreed. I usually don't use switch unless there are lots of same if's.